transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystem14Parser.java

```diff
@@ -110,6 +113,11 @@ public void readElement(XMLExtendedStreamReader reader, List<ModelNode> list) th
             }
             readElement(reader, element, list, subsystem, logStoreOperation);
         }
+
+        if(!relativeToEncountered) {
+            TransactionSubsystemRootResourceDefinition.OBJECT_STORE_RELATIVE_TO.parseAndSetParameter("jboss.server.data.dir", subsystem, reader);
```

Comment:
This logic should not execute if it's the subclass TransactionSubsystem31Parser running (or any future parser subclassed from this.) In the 3.1 xsd there is no longer this default.